internal/processor/filtermatcher/filtermatcher_test.go

```diff
@@ -0,0 +1,399 @@
+	return &filterset.Config{
+		MatchType: matchType,
+	}
+}
+
+func TestSpan_validateMatchesConfiguration_InvalidConfig(t *testing.T) {
+	version := "["
+	testcases := []struct {
+		name        string
+		property    filterconfig.MatchProperties
+		errorString string
+	}{
+		{
+			name: "regexp_match_type_for_int_attribute",
+			property: filterconfig.MatchProperties{
+				Config: *createConfig(filterset.Regexp),
+				Attributes: []filterconfig.Attribute{
+					{Key: "key", Value: 1},
+				},
+			},
+			errorString: `error creating attribute filters: match_type=regexp for "key" only supports STRING, but found INT`,
+		},
+		{
+			name: "unknown_attribute_value",
+			property: filterconfig.MatchProperties{
+				Config: *createConfig(filterset.Strict),
+				Attributes: []filterconfig.Attribute{
+					{Key: "key", Value: []string{}},
+				},
+			},
+			errorString: `error creating attribute filters: error unsupported value type "[]string"`,
+		},
+		{
+			name: "invalid_regexp_pattern_attribute",
+			property: filterconfig.MatchProperties{
+				Config:     *createConfig(filterset.Regexp),
+				SpanNames:  []string{"["},
+				Attributes: []filterconfig.Attribute{{Key: "key", Value: "["}},
+			},
+			errorString: "error creating attribute filters: error parsing regexp: missing closing ]: `[`",
+		},
+		{
+			name: "invalid_regexp_pattern_resource",
+			property: filterconfig.MatchProperties{
+				Config:    *createConfig(filterset.Regexp),
+				Resources: []filterconfig.Attribute{{Key: "key", Value: "["}},
+			},
+			errorString: "error creating resource filters: error parsing regexp: missing closing ]: `[`",
+		},
+		{
+			name: "invalid_regexp_pattern_library_name",
+			property: filterconfig.MatchProperties{
+				Config:    *createConfig(filterset.Regexp),
+				Libraries: []filterconfig.InstrumentationLibrary{{Name: "["}},
+			},
+			errorString: "error creating library name filters: error parsing regexp: missing closing ]: `[`",
+		},
+		{
+			name: "invalid_regexp_pattern_library_version",
+			property: filterconfig.MatchProperties{
+				Config:    *createConfig(filterset.Regexp),
+				Libraries: []filterconfig.InstrumentationLibrary{{Name: "lib", Version: &version}},
+			},
+			errorString: "error creating library version filters: error parsing regexp: missing closing ]: `[`",
+		},
+		{
+			name: "empty_key_name_in_attributes_list",
+			property: filterconfig.MatchProperties{
+				Config:   *createConfig(filterset.Strict),
+				Services: []string{"a"},
+				Attributes: []filterconfig.Attribute{
+					{
+						Key: "",
+					},
+				},
+			},
+			errorString: "error creating attribute filters: can't have empty key in the list of attributes",
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			output, err := NewMatcher(&tc.property)
+			assert.Zero(t, output)
+			assert.EqualError(t, err, tc.errorString)
+		})
+	}
+}
+
+func TestSpan_Matching_False(t *testing.T) {
```

Comment:
```suggestion
func Test_Matching_False(t *testing.T) {
```
Please also remove the "Span" from the names of the tests that are generic filter tests and not span-specific.

Reply:
> @zeitlinger lgtm, except the minor comment about usage of "Span" in the extracted tests. Please fix and can merge.

done